RCNN_tf.py

```diff
@@ -116,209 +176,292 @@ def loop_body(x, recurrent_states, output):
+    """Computes the accuracy of the classifier
+
+    Args:
+        labels  (tf.Tensor): the true labels
+        results (tf.Tensor): the predicted labels
+
+    Returns:
+        acc     (tf.Tensor): the accuracy of the classifier
+    """
     correct_prediction = tf.equal(tf.argmax(labels, 1), tf.argmax(result, 1))
-    return tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
+    acc = tf.reduce_mean(tf.cast(correct_prediction, tf.float32))
+
+    return acc
 
 
 class RCNN:
+    """The Recurrent Convolutional Neural Network Classifier
+
+    Attributes:
+        input_shape        (list): specification of the image input shape
+        output_shape       (list): specification of the class output shape
+        processing_unit     (str): specification of the PU to be used
+        learning_rate     (float): learning rate
+        num_filter          (int): number of filters to be used
+        shuf_buf_size       (int): buffer size for shuffling
+        conv_filter_shape  (list): specification of conv filter shape
+        rconv_filter_shape (list): specification of recurrent conv filter shape
+        pool_shape         (list): specification of pooling shape
+        pool_stride_shape  (list): spec of pooling stride shape
+    """
     def __init__(
             self,
             input_shape,
-            device_name="/cpu:0",
             output_shape=[None, 10],
+            processing_unit="/cpu:0",
             learning_rate=.1,
             num_filter=64,
-            shuffle_buffer_size=10000,
-            conv_layer_filter_shape=[5, 5],
-            rec_conv_layer_filter_shape=[3, 3],
-            pooling_shape=[3, 3],
-            pooling_stride_shape=[2, 2]
-    ):
+            shuf_buf_size=10000,
+            conv_filter_shape=[5, 5],
+            rconv_filter_shape=[3, 3],
+            pool_shape=[3, 3],
+            pool_stride_shape=[2, 2]
+        ):
+
+        # Input and output shape placeholders
         self.input_placeholder = tf.placeholder(PRECISION_TF, input_shape)
         self.output_placeholder = tf.placeholder(PRECISION_TF, output_shape)
 
-        self.device_name = device_name
+        # Processing unit placeholder
+        self.processing_unit = processing_unit
 
-        # dropout probability placeholder
+        # Dropout probability placeholder
         self.rate_placeholder = tf.placeholder(PRECISION_TF)
-        # number of data placeholder
+
+        # Number of data placeholder
         self.num_data_placeholder = tf.placeholder(tf.int64)
 
         # Create data set objects
         training_data_set = tf.data.Dataset.from_tensor_slices((
             self.input_placeholder,
-            self.output_placeholder
-        )).shuffle(buffer_size=shuffle_buffer_size).repeat().batch(batch_size=self.num_data_placeholder)
+            self.output_placeholder)
+            ).shuffle(buffer_size=shuf_buf_size
+            ).repeat(
+            ).batch(batch_size=self.num_data_placeholder)
```

Comment:
This line splitting is very hard to read I find

Reply:
Yeah, I agree: change it if you have a better formatting in mind.

In Javascript I would write something like:

```
self.output_placeholder))
     .shuffle(buffer_size=shuf_buf_size)
     .repeat()
     .batch(batch_size=self.num_data_placeholder)
```

Unfortunately Python cannot parse that....